Signal/src/ViewControllers/ConversationView/Cells/ConversationMediaView.swift

```diff
@@ -48,13 +62,53 @@ public class ConversationMediaView: UIView {
         } else {
             // TODO: Handle this case.
             owsFailDebug("Attachment has unexpected type.")
+            configureForMissingOrInvalid()
         }
     }
 
-    private func addMediaSubview(_ subview: UIView) {
-        addSubview(subview)
-        subview.autoPinEdgesToSuperviewEdges()
-        // TODO: Possibly add upload/download indicator here.
+    //
+    typealias ProgressCallback = (Bool) -> Void
+
+    private func addDownloadProgressIfNecessary() {
+        guard let attachmentPointer = attachment as? TSAttachmentPointer else {
+            owsFailDebug("Attachment has unexpected type.")
+            configureForMissingOrInvalid()
+            return
+        }
+        guard let attachmentId = attachmentPointer.uniqueId else {
+            owsFailDebug("Attachment stream missing unique ID.")
+            configureForMissingOrInvalid()
+            return
+        }
+
+        guard let progress = attachmentDownloads.downloadProgress(forAttachmentId: attachmentId) else {
```

Comment:
Seems like we're not doing anything with this `progress`?

Reply:
Modified guard.